strawberryfields/backends/tfbackend/circuit.py

```diff
@@ -38,6 +38,13 @@
 from scipy.special import factorial
 import tensorflow as tf
 
+try:
+    from tensorflow.python.ops.special_math_ops import _einsum_v1
+
+    tf.einsum = _einsum_v1
+except ImportError:
+    pass
```

Comment:
With TF 2.1, the legacy `tf.einsum` was renamed to `_einsum_v1`, while the replacement `tf.einsum` introduced the bug. This `try-except` block will dynamically patch versions where `_einsum_v1` exists, to make it the default einsum implementation _within this file only_.